code/modules/mining/minebot.dm

```diff
@@ -100,21 +157,118 @@
-		return
-	if(I.tool_behaviour == TOOL_CROWBAR || istype(I, /obj/item/borg/upgrade/modkit))
-		I.melee_attack_chain(user, stored_gun, params)
-		return
+	if(maintance_hatch_open)
+		if(I.tool_behaviour == TOOL_CROWBAR || istype(I, /obj/item/borg/upgrade/modkit))
+			I.melee_attack_chain(user, stored_gun, params)
+			scanner.forceMove(get_turf(src))
+			cutter.forceMove(get_turf(src))
+			return
+		else if(istype(I, /obj/item/stack/cable_coil) && health < maxHealth)
+			to_chat(user, span_notice("You begin replacing broken wires on [src]..."))
+			repair_burn(I, user)	
+			return	
+		else if(istype(I, /obj/item/gun/energy/plasmacutter))
+			if(cutter)
+				to_chat(user, span_notice("You replace [src]'s plasmacutter with [I]..."))
+				cutter.forceMove(get_turf(user))
+				cutter = null
+			else
+				to_chat(user, span_notice("You insert [I] into the plasma cutter mount on the [src]..."))
+			I.forceMove(src)
+			cutter = I
+			return
+		else if(istype(I, /obj/item/stack/ore/plasma) && cutter)
+			cutter.attackby(I)
+			if(cutter.cell.charge == cutter.cell.maxcharge) 
+				I.forceMove(src)
+			return		
+		else if(istype(I, /obj/item/gun/energy/kinetic_accelerator/mega))
+			equip_gun(I)
+			return
+		else if(istype(I, /obj/item/reagent_containers/hypospray/medipen))
+			if(pen)
+				to_chat(user, span_notice("You replace [src]'s medipen with a new one."))
+				to_chat(src, span_notice("[user] replaces your current medipen with a new one."))
+				pen.forceMove(get_turf(src))
+			else
+				to_chat(user, span_notice("You insert a medipen into [src]."))
+				to_chat(src, span_notice("[user] inserts a new medipen into you."))
+			I.forceMove(src)
+			pen = I
+			return
+		else if(istype(I, /obj/item/wormhole_jaunter))
+			if(jaunter)
+				to_chat(user, span_warning("[src] already has a jaunter!"))
+				return
+			to_chat(user, span_notice("You install a jaunter into [src]!"))
+			I.forceMove(src)
+			jaunter = I
+			return
+		else if(istype(I, /obj/item/bikehorn))
+			if(honk)   /// HONK HONK HONK HONK HONK XDXDXDXD 
+				to_chat(user, span_warning("[src] already has a bike horn!"))
+				return
+			to_chat(user, span_notice("You install a bike horn into [src]!"))
+			I.forceMove(src)
+			honk = I
+			playsound(src, 'sound/items/bikehorn.ogg', 50, TRUE)
+			return
+		else if(istype(I, /obj/item/t_scanner/adv_mining_scanner/))
+			if(scanner)
+				to_chat(user, span_notice("You replace [src]'s mining scanner with a new one."))
+				to_chat(src, span_notice("[user] replaces your current mining scanner with a new one."))
+				scanner.forceMove(get_turf(src))
+			else
+				to_chat(user, span_notice("You insert a mining scanner into [src]."))
+				to_chat(src, span_notice("[user] inserts a new mining scanner into you."))
+			I.forceMove(src)
+			scanner = I
+			return
+		else if(istype(I, /obj/item/pen && user.a_intent == INTENT_HELP))  //So we can get attacked by edaggers
+			var/new_name = input(user, "Choose a new name for the minig bot", "Name")
```

Comment:
Sanitize this and run it through the pretty filter, always do that with user input.